### PR TITLE
Prepare for release v1.2.0-rc1

### PR DESCRIPTION
**Pull request by ianvernon**
Signed-off by: Ian Vernon <ian@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5035)

### CODE REVIEW

**Comment by ianvernon** [comment]
This is waiting on #5034 .

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-missed-k8s

**Comment by ianvernon** [comment]
test-upstream-k8s

**Review by eloycoto — APPROVED** [review]

**Review by raybejjani — APPROVED** [review]
approved for k8s

File: examples/kubernetes/1.10/cilium-ds.yaml [hunk 1]
```diff
@@ -51,7 +51,7 @@ spec:
                   optional: true
                   key: clean-cilium-state
       containers:
-        - image: docker.io/cilium/cilium:latest
+        - image: docker.io/cilium/cilium:v1.2.0-rc1
           imagePullPolicy: Always
```
Comment (aanm) [inline examples/kubernetes/1.10/cilium-ds.yaml]: `imagePullPolicy: IfNotPresent` for all k8s versions. Step 5 of the release process http://cilium.readthedocs.io/en/latest/contributing/?#generic-release-process

**Review by aanm — DISMISSED** [review]

**Comment by tgraf** [comment]
@ianvernon Let's pull in the latest backports that have been merged

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-upstream-k8s

**Comment by ianvernon** [comment]
test-missed-k8s

File: examples/kubernetes/1.10/cilium-ds.yaml (hunk 1 above)
Comment (ianvernon) [reply examples/kubernetes/1.10/cilium-ds.yaml]: Thanks for catching this, I somehow reversed the two in my head when making the release on Friday.

**Comment by ianvernon** [comment]
test-docs-please

**Review by aanm — APPROVED** [review]

**Review by aanm — CHANGES_REQUESTED** [review]
The NEWS.rst needs to be changed with a `*` on each line. Check the previous releases. https://github.com/cilium/cilium/blob/cc13ec1a9c36cded7e492d1893cfb9e6bfc962d6/NEWS.rst

**Comment by ianvernon** [comment]
Sigh... that really should be part of the generation script. I'll make an issue to track that and fix it manually, locally...

**Comment by eloycoto** [comment]
LGTM

**Comment by ianvernon** [comment]
@aanm the formatting was messed up, but the fix is different; it now uses:

```
::
    André Martins (93):
        examples/kubernetes:...
```

The `::`  and 4 spaces indentation were missing. This was my mistake.

**Comment by ianvernon** [comment]
All tests passed (docs, upstream K8s, missed K8s, and normal CI tests). There were no more code changes to this latest upload apart from the `NEWS.rst` fixes so I don't see a need to re-run the tests.

**Review by raybejjani — APPROVED** [review]

**Review by tgraf — APPROVED** [review]

**Pull request merged**